src/Stratis.Bitcoin/Utilities/DBreezeSerializer.cs

```diff
@@ -24,28 +27,25 @@ public DBreezeSerializer(Network network)
         /// <returns>Binary data representing the serialized object.</returns>
         public byte[] Serialize(object obj)
         {
-            var serializable = obj as IBitcoinSerializable;
-            if (serializable != null)
+            if (obj is IBitcoinSerializable serializable)
                 return serializable.ToBytes(this.Network.Consensus.ConsensusFactory);
 
-            var u256 = obj as uint256;
-            if (u256 != null)
+            if (obj is uint256 u256)
                 return u256.ToBytes();
 
-            var u160 = obj as uint160;
-            if (u160 != null)
+            if (obj is uint160 u160)
                 return u160.ToBytes();
 
-            var u32 = obj as uint?;
-            if (u32 != null)
+            if (obj is uint u32)
                 return u32.ToBytes();
 
-            var arr = obj as object[];
-            if (arr != null)
+            if (obj is IEnumerable<object> collection)
```

Comment:
Is it always the case? `IEnumerable<object> ==object[]`

Reply:
`IEnumerable<object>` works with lists and hashsets. `object[]` works only with arrays. 
So because it wasn't the case I've changed this line